### PR TITLE
add detailed TiDBCluster example

### PR DESCRIPTION
**Pull request by zjj2wry**
### What problem does this PR solve? 
Fixes https://github.com/pingcap/tidb-operator/issues/2440

### Check List 

Tests 
 - Manual test (add detailed scripts or steps below)

Related changes
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
@YangKeao ready for review, thanks

File: examples/detailed-tidb/tidb-cluster.yaml [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+  #   config:
+  #     addr: 0.0.0.0:8250
+  #     gc: 7
+  #     heartbeat-interval: 2
+
+  ## Ref: TiCDC is a tool for replicating the incremental data of TiDB
+  ## Ref: https://pingcap.com/docs/tidb-in-kubernetes/stable/deploy-ticdc/
+  # ticdc:
+  #   baseImage: pingcap/ticdc
+  #   replicas: 3
```
Comment (Yisaer) [inline examples/detailed-tidb/tidb-cluster.yaml:208]: I think we don't need to expose `cdc` and `binlog` in a detailed tidbcluster, there should be treated as a independent case.

**Comment by cofyc** [comment]
I would prefer using `advanced` as the directory name. Compared to basic examples, yamls in  `advanced` can be used as examples in more advanced scenarios.

File: examples/detailed-tidb/tidb-cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+      cpu: "100m"
+      storage: 1Gi
+
+    ## The storageClassName of the persistent volume for TiKV data storage.
+    ## Defaults to Kubernetes default storage class.
+    # storageClassName: ""
+
+    ## Affinity for pod assignment
+    ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
+    # affinity:
```
Comment (zjj2wry) [inline examples/detailed-tidb/tidb-cluster.yaml:160]: @cofyc @Yisaer Do we need to turn on affinity？

Comment (zjj2wry) [reply examples/detailed-tidb/tidb-cluster.yaml:160]: tidb-schedule already guarantees high availability

File: examples/detailed-tidb/tidb-cluster.yaml (hunk 1 above)
Comment (zjj2wry) [reply examples/detailed-tidb/tidb-cluster.yaml:208]: actually these features are not used, now compared to basic tidb cluster just add resource.request.

**Comment by zjj2wry** [comment]
> I would prefer using advanced as the directory name. Compared to basic examples, yamls in advanced can be used as examples in more advanced scenarios.

done

**Pull request merged**